components/eamxx/src/physics/mam/eamxx_mam_aci_process_interface.cpp

```diff
@@ -0,0 +1,1299 @@
+  EKAT_REQUIRE_MSG(
+      used_mem == requested_buffer_size_in_bytes(),
+      "Error! Used memory != requested memory for MAMMicrophysics.");
+}  // function init_buffers ends
+
+// ================================================================
+//  INITIALIZE_IMPL
+// ================================================================
+void MAMAci::initialize_impl(const RunType run_type) {
+  // ------------------------------------------------------------------------
+  // ## Runtime options
+  // ------------------------------------------------------------------------
+
+  wsubmin_ = m_params.get<double>("wsubmin");
+  top_lev_ = m_params.get<int>("top_level_mam4xx");
+
+  // ------------------------------------------------------------------------
+  // Input fields read in from IC file, namelist or other processes
+  // ------------------------------------------------------------------------
+  w_sec_mid_    = get_field_in("w_variance").get_view<const Real **>();
+  dgnum_        = get_field_in("dgnum").get_view<const Real ***>();
+  liqcldf_      = get_field_in("cldfrac_liq").get_view<const Real **>();
+  liqcldf_prev_ = get_field_in("cldfrac_liq_prev").get_view<const Real **>();
+  kvh_mid_      = get_field_in("eddy_diff_heat").get_view<const Real **>();
+
+  // store fields only to be converted to dry mmrs in wet_atm_
+  wet_atm_.qv = get_field_in("qv").get_view<const Real **>();
+  wet_atm_.qc = get_field_in("qc").get_view<const Real **>();
+  wet_atm_.nc = get_field_in("nc").get_view<const Real **>();
+  wet_atm_.qi = get_field_in("qi").get_view<const Real **>();
+  wet_atm_.ni = get_field_in("ni").get_view<const Real **>();
+
+  // store rest fo the atm fields in dry_atm_in
+  dry_atm_.T_mid = get_field_in("T_mid").get_view<const Real **>();
+  dry_atm_.p_mid = get_field_in("p_mid").get_view<const Real **>();
+  dry_atm_.p_int = get_field_in("p_int").get_view<const Real **>();
+  dry_atm_.p_del = get_field_in("pseudo_density").get_view<const Real **>();
+  dry_atm_.omega = get_field_in("omega").get_view<const Real **>();
+
+  // store fields converted to dry mmr from wet mmr in dry_atm_
+  dry_atm_.qv = buffer_.qv_dry;
+  dry_atm_.qc = buffer_.qc_dry;
+  dry_atm_.nc = buffer_.nc_dry;
+  dry_atm_.qi = buffer_.qi_dry;
+  dry_atm_.ni = buffer_.ni_dry;
+
+  // pbl_height
+  dry_atm_.pblh = get_field_in("pbl_height").get_view<const Real *>();
+
+  // geometric thickness of layers (m)
+  dry_atm_.dz = buffer_.dz;
+
+  // geopotential height above surface at interface levels (m)
+  dry_atm_.z_iface = buffer_.z_iface;
+
+  // geopotential height above surface at mid levels (m)
+  dry_atm_.z_mid = buffer_.z_mid;
+
+  // total cloud fraction
+  dry_atm_.cldfrac = get_field_in("cldfrac_tot").get_view<const Real **>();
+
+  // computed updraft velocity
+  dry_atm_.w_updraft = buffer_.w_updraft;
+
+  // ------------------------------------------------------------------------
+  // Output fields to be used by other processes
```

Comment:
It would be nice to have the CCN diagnostic output fields (e.g., CCN@0.1%, 0.2%, and 0.5%). Perhaps this can be addressed in a future PR.

Reply:
Yes, we can do that as a diagnostics in a future PR.